File: 167. Two Sum II - Input Array Is Sorted/twoSum2.py
'''
    LeetCode problem 167. Two Sum II - Input Array Is Sorted
    https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
'''

from typing import List


class Solution:
    def twoSum2(self, numbers: List[int], target: int) -> List[int]:
        """
        Time complexity: O(n)
        Space complexity: O(n)
        """
        if numbers == [] or numbers is None:
            length = 0
        else:
            length = len(numbers)
        found_values = dict()
        for i in range(length):
            if numbers[i] not in found_values:
                found_values[numbers[i]] = i
            wanted = target - numbers[i]
            if wanted in found_values and i != found_values[wanted]:
                return [found_values[wanted] + 1, i + 1]

File: 167. Two Sum II - Input Array Is Sorted/cs/TwoSum2/TwoSum2.cs
/*
    LeetCode problem 167. Two Sum II - Input Array Is Sorted
    https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
*/

namespace LeetCode.TwoSum2;

public class Solution()
{
    public int[] TwoSum2(int[] numbers, int target)
    {
        int left = 0, right = numbers.Length - 1;
        while (left < right)
        {
            int s = numbers[left] + numbers[right];
            if (s == target)
                return [left + 1, right + 1];
            else if (s < target)
                left += 1;
            else
                right -= 1;
        }
        throw new InvalidOperationException("No solution found.");
    }
}